### ISSUE TITLE
Deprecating OpenWithOptions

### ISSUE THREAD

**Issue by asdine** [asdine/storm#14]
Using something like this would make the `DB` creation more consistent:

```go
db := storm.Open("my.db", storm.BoltOptions(0600, &bolt.Options{Timeout: 1 * time.Second}))
```

**Comment by bep** [comment]
On a slightly related topic:

The signature for the config funcs should maybe be:

```
func(*DB) error
```

None of the current config funcs can fail, but it might be smart to get the signature in place before people start making up their own.

**Comment by asdine** [comment]
Fixed by bac796ec5c703999108d34c87dfa12a242fdcbf8
